src/common/host_device_vector.h

```diff
@@ -250,11 +250,19 @@ class HostDeviceVector {
 
   /*!
    * \brief Specify memory distribution.
-   *
-   *   If GPUSet::Empty() is used, all data will be drawn back to CPU.
    */
-  void Reshard(const GPUDistribution& distribution) const;
-  void Reshard(GPUSet devices) const;
+  void Shard(const GPUDistribution &distribution) const;
+  void Shard(GPUSet devices) const;
+
+  /*!
+   * \brief Pull data back to CPU.
+   */
+  void Unshard() const;
```

Comment:
Do we really need this?

To pull the data to the host, use `ConstHostVector()`. `HostVector()` (for a non-const HostDeviceVector) additionally invalidates the device data (but doesn't free memory).

If resetting the distribution is needed, `Reshard(GPUDistribution::Empty())` should be able to do it.

Reply:
Removed